syncer/syncer.go

```diff
@@ -1308,9 +1316,53 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		failpoint.Inject("ProcessBinlogSlowDown", nil)
 
 		s.tctx.L().Debug("receive binlog event", zap.Reflect("header", e.Header))
+
+		startLocation = currentLocation.Clone()
+		// TODO: support all event
+		// we calculate endLocation(currentLocation) for Rows/Query event here
+		switch e.Event.(type) {
+		case *replication.RowsEvent, *replication.QueryEvent:
+			endSuffix := startLocation.Suffix
+			if s.firstReplace {
```

Comment:
how about using `inReplaceError` or something, to replace

1. here and in `(*Syncer).getEvent` we only check `inReplaceError` instead of `firstReplace` and `Suffix`
2. if `inReplaceError` we don't call `s.errOperatorHolder.Apply`, so in `s.errOperatorHolder.Apply` don't need to check `endLocation.Suffix != 0
3. set false when `currentLocation.Suffix = 0`, set true when `s.firstReplace = true`

Reply:
> here and in (*Syncer).getEvent we only check inReplaceError instead of firstReplace and Suffix

in shardingResync, maybe suffix is not zero but inReplaceError is false?

Reply:
I made these modification to substitude `inReplaceError` for `firstReplace`, just write here for your reference.

- when enter `(*Syncer).Run` (both start and resume), set `inReplaceError = false`
- in `closeShardingResync`, set `inReplaceError` according to new `currentLocation.Suffix`
- in start sharding re-sync logic, set `inReplaceError` according to new `currentLocation.Suffix`
- when `currentLocation.Suffix = 0`, set `inReplaceError = false`
- when originally `firstReplace = true`, set `inReplaceError = true`

so the benifit is 
- when we need to check `firstReplace` and `e.Suffix`, now only need to check this one variable
- if `inReplaceError` we don't enter `s.errOperatorHolder.Apply`, so in `s.errOperatorHolder.Apply` don't need to check `endLocation.Suffix != 0`
- maybe easier to understand running flow 😂 

don't know if worthy changing

Reply:
address in 0018b5d